File: seoyepar/20230103/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(int n, int size, vector<vector<int>> computers, int *visited)
{
    visited[n] = 1;
    for (int i = 0; i < size; i++)
    {
        if (computers[n][i] && !visited[i])
            dfs(i, size, computers, visited);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    int *visited = new int[computers.size()]{};
    
    for (int i = 0; i < computers.size(); i++)
    {
         if (visited[i] == 0)
         {
             dfs(i, computers[i].size(), computers, visited);
             answer++;
         }
    }
    return answer;
}

File: seoyepar/20230414/네트워크.js
function solution(n, computers) {
    let visited = new Array(n).fill(0);
    let cnt = 0;
    
    function dfs(pos) {
        if (!visited[pos]){
            visited[pos] = 1;
            computers[pos].forEach((data, i)=>{
                if (data === 1) dfs(i);
            });
        }
    }
    for (let i = 0; i < n; i++){
        if(!visited[i]){
            dfs(i);
            cnt++;
        }
    }
       
    return cnt;
}
